src/groups/mqb/mqbstat/mqbstat_brokerstats.h

```diff
@@ -159,6 +166,42 @@ inline bmqst::StatContext* BrokerStats::statContext()
     return d_statContext_p;
 }
 
+template <>
+inline void
+BrokerStats::onEvent<BrokerStats::EventType::Enum::e_CLIENT_CREATED>()
+{
+    BSLS_ASSERT_SAFE(d_statContext_p && "initialize was not called");
```

Comment:
Not in the scope of this PR, but It's a pity that we check this condition on every metric update. Would be nice to ensure initialization on construction, so `Stats` classes are impossible to construct without this.